src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/AssetsController.cs

```diff
@@ -78,5 +80,68 @@ public async Task<IActionResult> GetAsset(int id)
 
             return Ok(new Asset(asset));
         }
+
+        [HttpPost("{assetId}/locations")]
+        [SwaggerResponse((int)HttpStatusCode.Created, Type = typeof(Channel))]
+        public async Task<IActionResult> AddAssetLocationToAsset(int assetId, [Required] string location, [Required] LocationType assetLocationType)
+        {
+            var assetLocation = new Data.Models.AssetLocation
+            {
+                Location = location,
+                Type = (Maestro.Data.Models.LocationType) assetLocationType,
+            };
+
+            Maestro.Data.Models.Asset asset = await _context.Assets
+                .Include(a => a.Locations)
+                .Where(a => a.Id == assetId)
+                .SingleOrDefaultAsync();
+
+            if (asset == null)
+            {
+                return NotFound(new ApiError($"The asset with id '{assetId}' was not found."));
+            }
+
+            // If asset location is already in asset, nothing to do
+            if (asset.Locations != null &&
+                asset.Locations.Any(existing => existing.Location.Equals(assetLocation.Location, StringComparison.OrdinalIgnoreCase) &&
+                existing.Type == assetLocation.Type))
+            {
+                return StatusCode((int)HttpStatusCode.NotModified);
+            }
+
+            asset.Locations = asset.Locations ?? new List<Data.Models.AssetLocation>();
+            asset.Locations.Add(assetLocation);
+
+            await _context.SaveChangesAsync();
+            return StatusCode((int)HttpStatusCode.Created);
```

Comment:
This should probably return the AssetLocation object that was added.

Reply:
will fix.